snap/info.go

```diff
@@ -644,36 +644,28 @@ func (s *Info) CommonDataSaveDir() string {
 	return CommonDataSaveDir(s.InstanceName())
 }
 
-// Get home directories from system-params file.
-// Home directories are returned as a ',' seperated string
-func HomeDirsSysParams(rootDir string) (string, error) {
-
-	ssp, err := sysparams.Open(rootDir)
-	if err != nil {
-		return "", err
-	}
-
-	return ssp.Homedirs, nil
-}
-
-// DataHomeGlob returns the globbing expression for the snap directories in use
-func DataHomeGlob(opts *dirs.SnapDirOptions) []string {
+// DataHomeGlobs returns a slice of globbing expressions for the snap directories in use
+func DataHomeGlobs(opts *dirs.SnapDirOptions) []string {
 
 	if opts == nil {
 		opts = &dirs.SnapDirOptions{}
 	}
 
-	homedirs, err := HomeDirsSysParams(dirs.GlobalRootDir)
-
-	// if we cannot read the file or if the homedirs value is empty we use dirs defaults
-	if err != nil || homedirs == "" {
+	// If we cannot read the file or if the homedirs value is empty we use dirs defaults
+	if dirs.SnapHomeDirs == "" {
 		if opts.HiddenSnapDataDir {
 			return []string{dirs.HiddenSnapDataHomeGlob}
 		}
 		return []string{dirs.SnapDataHomeGlob}
 	}
 
-	list := strings.Split(homedirs, ",")
+	list := strings.Split(dirs.SnapHomeDirs, ",")
```

Comment:
I think this list contains the base home directories and not the snap data dir globs.

I believe each entry should be something like this:
```golang
if opts.HiddenSnapDataDir {
    filepath.Join(baseHomeDir, "*", HiddenSnapDataHomeDir)
} else {
    filepath.Join(baseHomeDir, "*", UserHomeSnapDir)
}
```

Reply:
That is correct, each element in the list is later converted to be a globbing expression towards the end of the function in a for loop.


		if opts.HiddenSnapDataDir {
			list[i] = dir + "/*/" + dirs.HiddenSnapDataHomeDir
		} else {
			list[i] = dir + "/*/" + dirs.UserHomeSnapDir
		}

Reply:
my bad, I was seeing the changes made by the last commit only by mistake.